src/common/queries/group-settings.ts

```diff
@@ -10,10 +10,55 @@
 
 import { endpoint } from '$app/common/helpers';
 import { request } from '$app/common/helpers/request';
-import { useQuery } from 'react-query';
+import { useQuery, useQueryClient } from 'react-query';
+import { route } from '../helpers/route';
+import { GenericSingleResourceResponse } from '../interfaces/generic-api-response';
+import { GroupSettings } from '../interfaces/group-settings';
+import { toast } from '../helpers/toast/toast';
 
 export function useGroupSettingsQuery() {
   return useQuery('/api/v1/group_settings', () => {
     return request('GET', endpoint('/api/v1/group_settings'));
   });
 }
+
+interface Params {
+  id: string | undefined;
+}
+
+export const useGroupQuery = (params: Params) => {
```

Comment:
`function` style of declaration, please.

Reply:
Changed!